src/ServerRequestFactoryInterface.php

```diff
@@ -12,20 +12,9 @@
      *
      * @param string $method
      * @param UriInterface|string $uri
+     * @param array $server optional server params ($_SERVER or similar structure)
      *
      * @return ServerRequestInterface
      */
-    public function createServerRequest($method, $uri);
-
-    /**
-     * Create a new server request from server variables.
-     *
-     * @param array $server Typically $_SERVER or similar structure.
-     *
-     * @return ServerRequestInterface
-     *
-     * @throws \InvalidArgumentException
-     *  If no valid method or URI can be determined.
-     */
-    public function createServerRequestFromArray(array $server);
+    public function createServerRequest($method, $uri, $server = []);
```

Comment:
@mindplay-dk Threating server-params, URI and method as distinct model values seems resonable. However, I think this would require stating it explicitly in the method because it is far from obvious.